src/krb5_plugin/sssd_krb5_locator_plugin.c

```diff
@@ -481,20 +484,24 @@ krb5_error_code sssd_krb5_locator_lookup(void *private_data,
             return KRB5_PLUGIN_NO_HANDLE;
         }
 
-        if (svc == locate_service_kadmin || svc == locate_service_kpasswd ||
-            svc == locate_service_master_kdc) {
-            ret = get_krb5info(realm, ctx, locate_service_kpasswd);
+    }
+
+    if (ctx->kpasswd_addr == NULL
```

Comment:
I don't understand this change. Can `ctx->kpasswd_addr` ever be non-NULL? Previously the code calls ` get_krb5info(realm, ctx, locate_service_kdc);` which will only initialize ctx->kdc_addr. Is this extra safety measure for future?

Reply:
This was discovered by the test I added.

In the older version the files were read based on the first request to the plugin, i.e. if it was a request for a KDC the kpasswdinfo file was never read even if it existed. Since the caller will load the plugin only once a long running process might first try to find a KDC and later a server for a password change. In this case ctx->kpasswd_addr would have been NULL in the old code.

With this change kpasswdinfo will be read in this case or if it does not exist the KDC data is copied.